Redundant double precision multiply

I habitually compile with  -Wdouble-promotion to catch inadvertant double precision operations where float is just fine. I recently started to see the warnings below. Since the value returned is a float I can't see any benefit from a double precision multiply. The obvious fix is to define S322F_SCALE as a float constant.

../Daisy/libdaisy/src/daisy_core.h: In function 'float s322f(int32_t)':
../Daisy/libdaisy/src/daisy_core.h:96:21: warning: implicit conversion from 'float' to 'double' to match other operand of binary expression [-Wdouble-promotion]
   96 |     return (float)x * S322F_SCALE;

--- a/src/daisy_core.h
+++ b/src/daisy_core.h
@@ -37,7 +37,7 @@ cache enabled.
 #define F2S24_SCALE 8388608.0f        /**< 2 ** 23 */
 #define S242F_SCALE 1.192092896e-07f  /**< 1 / (2 ** 23) */
 #define S24SIGN 0x800000              /**< 2 ** 23 */
-#define S322F_SCALE 4.6566129e-10     /**< 1 / (2** 31) */
+#define S322F_SCALE 4.6566129e-10f    /**< 1 / (2** 31) */
 #define F2S32_SCALE 2147483647.f      /**< (2 ** 31) - 1 */
 
 /** 

